qiskit/synthesis/arithmetic/adders/cdkm_ripple_carry_adder.py

```diff
@@ -0,0 +1,154 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Compute the sum of two qubit registers using ripple-carry approach."""
+
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.circuit.quantumregister import QuantumRegister, AncillaRegister
+
+
+def adder_ripple_c04(num_state_qubits: int, kind: str = "half") -> QuantumCircuit:
+    r"""A ripple-carry circuit to perform in-place addition on two qubit registers.
+
```

Comment:
This is nice. The actual synthesis functions still take the adder type as the parameter (i.e. we do not have individual functions `half_adder_ripple_c04`, `modular_adder_ripple_c04`, `full_adder_ripple_c04`).